src/main/java/org/jboss/ejb/client/legacy/SecurityUtils.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2017 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.ejb.client.legacy;
+
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+
+/**
+ * Security sensitive operations being used in this package.
+ *
+ * @author <a href="mailto:ropalka@redhat.com">Richard Opálka</a>
+ */
+final class SecurityUtils {
+
+    private SecurityUtils() {
+        // forbidden instantiation
+    }
+
+    static boolean getBoolean(final String propertyName) {
+        return getBoolean(propertyName, false);
+    }
+
+    static boolean getBoolean(final String propertyName, final boolean defaultValue) {
+        final SecurityManager sm = System.getSecurityManager();
```

Comment:
These calls should still be using the call to WildFlySecurityManager instead of directly using the AccessControler.

The WildFlySecurityManager contains an optimisation so when running within the application server with the security manager active we get to skip the overhead of a new protection domain.

Reply:
Proposal applied.